MetaMask/metamask-extension
# refactor: Bump `@metamask/providers` to `^16.0.0`.  Remove deprecated networkId

## **Description**

This PR bumps the `@metamask/providers` version, which removes properties that are not officially documented and should not have been exposed in the first place. It also removes support for the networkChange event and offline net_version resolution which will allow us to remove the last remaining chunk of networkId usage in Extension. These have all been properly deprecated, the most recent deprecations being 180 days old now.

Removes the following:
- window.ethereum.chainId
- window.ethereum.networkVersion
- window.ethereum.selectedAddress
- networkChanged event
- offline net_version support in window.ethereum.send()
- networkId resolution inside MetaMaskController

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23375?quickstart=1)

## **Related issues**

See: https://github.com/MetaMask/MetaMask-planning/issues/2195
See: https://github.com/MetaMask/metamask-improvement-proposals/discussions/23
See: https://github.com/MetaMask/MetaMask-planning/issues/2176
See: https://github.com/MetaMask/providers/pull/306


## **Manual testing steps**

1. Open a dapp
2. Connect to the dapp
3. In dapp web console, check the values of 
  - `window.ethereum.chainId`: should throw
  - `window.ethereum.selectedAddress`: should throw
  - `window.ethereum.networkVersion`: should be undefined
4. Open background extension network tab
  - in the dapp web console, `window.ethereum.send({method: 'net_version'}, console.log)`: should result in a network request to the rpc endpoint as this is no longer resolved locally/cached
5. In the dapp web console
  - `window.ethereum.on('networkChanged', console.log)`
  - In the wallet UI, switch networks
  - nothing should be logged

## **Screenshots/Recordings**



### **Before**

![Screenshot 2024-03-07 at 10 16 08 AM](https://github.com/MetaMask/metamask-extension/assets/918701/bc33c4e3-b9bd-495e-abb5-feab2148719d)

### **After**

![Screenshot 2024-03-07 at 10 23 54 AM](https://github.com/MetaMask/metamask-extension/assets/918701/8cb73ebb-4a18-457a-a048-288dea53bafa)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
